docs/source/dev/thrift_client.rst

```diff
@@ -0,0 +1,340 @@
+   <dependency>
+     <groupId>com.koverse</groupId>
+     <artifactId>koverse-thrift</artifact>
+     <version>2.9.0</version>
+   </dependency>
+ </dependencies>
+
+Configuring the Client
+----------------------
+
+The first step in developing software to use the client is to configure the client.
+That is done using the class ``com.koverse.client.thrift.ClientConfiguration``.
+This class uses the Builder object-oriented pattern.
+When the builder is created, all of the configuration properties are set to common defaults.
+Thus, it is not necessary to set a configuration parameter unless you need to change it from the default.
+
+.. code-block:: java
+
+ import com.koverse.client.thrift.ClientConfiguration;
+ ...
+ ClientConfiguration configuration = ClientConfiguration.builder().host("mykoverseServer.com").build();
+ ...
+
+You would of course specify a valid host name for your Koverse server instead of "myKoverseServer.com."
+Typically, the only configuration that needs to be changed is the host.
+There are several other configurations such as specific TCP/IP port numbers, but they are very rarely set to anything but their default values.
+
+Creating the Client
+-------------------
+
+The client is created by using one of two available constructors.
+Which constructor to use depends on how you need to authenticate with the Koverse server.
+There are two ways to authenticate: API Token or user name and password.
+For either constructor, the ``ClientConfiguration`` object must be passed in as well.
+
+Below is an example of authenticating with an API Token:
+
+.. code-block:: java
+
+ import com.koverse.client.thrift.Client;
+ ...
+ String apiToken = ...
+ Client client = new Client(configuration, apiToken);
+ ...
+
+And here is an example of authenticating with a user name and password:
+
+.. code-block:: java
+
+ import com.koverse.client.thrift.Client;
+ ...
+ String userName = ...
+ String password = ...
+ Client client = new Client(configuration, userName, password);
+ ...
+
+After constructing the client, you may use it to perform operations such as:
+
+* Perform queries (JSON object and Lucene based).
+* Get auto-suggest results for a query term.
+* Get a sample of records from a Data Set.
+* Retrieve Data Sets.
+* Create Data Sets
+* Retrieve API Tokens.
+* Create API Tokens.
+
+JSON Object Queries
+-------------------
+
+JSON object queries are specified as a Java String of JSON and has a similar syntax to that of MongoDB.
+The simplest type of object query is below, which simply searches for records that have any field with term "meow":
+
+.. code-block:: json
+
+ {
+   "$any" : "meow"
+ }
+
+The client can specify this query as a Java String object and a JSON framework can be used to construct the object.
+However, if a JSON framework is not being used and you are constructing the JSON string manually, care must be taken to format the JSON correctly.
+This is especially true when writing the quote character; it must be escaped.
```

Comment:
I don't think a semi-colon belongs here. Also, the below example shows double-quotes being escaped. Do single-quotes also need to be escaped? If not, the text should read double-quotes but even if it does you may want to specify both.